lib/iris/analysis/__init__.py

```diff
@@ -1228,6 +1228,15 @@ def _rms(array, axis, **kwargs):
     return rval
 
 
+def _lazy_rms(array, axis, **kwargs):
+    # XXX This should use `da.average` and not `da.mean`, as does the above.
+    # However `da.average` current doesn't handle masked weights correctly
+    # (see https://github.com/dask/dask/issues/3846).
+    # To work around this we use da.mean, which doesn't support weights at
+    # all, so we raise an error rather than silently giving the wrong answer.
```

Comment:
Where is the error being raised? Is that caused by passing invalid kwargs through to ``da.mean``?

Reply:
@pelson apologies, that comment isn't that clear on re-reading. But you're right, the error comes from `da.mean` receiving a bad kwarg:
```
TypeError: mean() got an unexpected keyword argument 'weights'
```
I'll improve the wording of the comment.